icechunk-python/python/icechunk/_icechunk_python.pyi

```diff
@@ -323,4 +326,50 @@ class StoreConfig:
         """Return the StoreConfig object as a JSON string"""
         ...
 
+class VersionSelection(Enum):
+    """Enum for selecting the which version of a conflict"""
+
+    Fail = 0
+    UseOurs = 1
+    UseTheirs = 2
+
+class ConflictSolver:
+    """An abstract conflict solver that can be used to detect or resolve conflicts between two stores
+
+    This should never be used directly, but should be subclassed to provide specific conflict resolution behavior
+    """
+
+    ...
+
+class BasicConflictSolver(ConflictSolver):
+    """A basic conflict solver that allows for simple configuration of resolution behavior"""
+
+    def __init__(
+        self,
+        *,
+        on_user_attributes_conflict: VersionSelection = VersionSelection.UseOurs,
+        on_chunk_conflict: VersionSelection = VersionSelection.UseOurs,
```

Comment:
Would `Fail` potentially be a better / safer default, requiring an explicit overwrite for a more aggressive strategy?

Reply:
The thing is, if you are going to say `Fail` everywhere, you shouldn't be calling `rebase` at all. In IC rebase is explicit, if you call it, you want to accept some kind of risk.

Reply:
Seba I'm not sure I follow. The current default Arraylake rebase behavior is to fail if there are any chunk conflicts or attribute conflicts. But you can still rebase, i.e. rewrite the snapshot lineage if there are no such conflicts.

This is desirable behavior in many scenarios and is a good default because it is the most "conservative" way to rebase.

Reply:
In fact we describe this exact scenario in the Arraylake docs: https://docs.earthmover.io/concepts/version-control-system#commit-process-and-conflict-resolution

Reply:
I think the behavior you want is supported by the `ConflictDetector` in Rust. That's also a `ConflictSolver` but it doesn't attempt to "fix" things, it just fails on any conflicts. I think using that solver instead, you can recover the AL behavior. At least that was my intention when I built it. 

Then the `BasicConflictSolver` uses `ConflictDetector` output and tries to fix it according to user policies, in a pretty naif way, but if you only want "fast forward if absolutely no conflicts", the detector should be enough.

I'm OK with any behavior we want to implement as a default, and I would also favor the safest one. That's why I made `rebase` explicit and not a default, unlike in AL. Any form of rebase implies risk. For example, there may be no conflicts, but you still computed chunk A from an old version of chunk B that was since then updated. If you never wrote to B, there are no conflicts but you end up with bad data anyway.

In my view, we should give the full menu of options available to Python users: fail on conflicts detected, basic resolution strategies, and, in the future, more advanced and even custom strategies.